### ISSUE TITLE
CORS blocks acces to "fonts.gstatic.com"

### ISSUE THREAD

**Issue by rpc-scandinavia** [SamProf/MatBlazor#269]
This is not directly an error, but a potential problem.
When I enable blocking of "Trackers" in Firefox, it also blocks the Google fonts.

**Reproduce:**
In Firefox, navigate to "Preferences -> Browser Privacy -> Content Blocking -> Custom" and select "Strict" or "Custom" with "Trackers" set to "In all windows".

**Error**
When I visit "https://www.matblazor.com/" or my own dev. test, all the icons are text insted of SVG. The MatBlazor site is really garbaled. In the console I see this error:

`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://fonts.gstatic.com/s/roboto/v20/KFOlCnqEu92Fr1MmEU9fBBc4.woff2. (Reason: CORS request did not succeed).`

**Why is this a problem?**
I just allowed trackers, and everything works.
But I think this is a problem, when people uses this build in feature - which I would prefere, if it was not for this problem.

**The solution**
I only see one solution.
That the developer can configure MatBlazor to load the fonts from the same server as the Blazor application - i.e. change the URI.

**Comment by chris579** [comment]
Best woud be to just remove any reference on third party fonts at all. The developer should be able to choose where the fonts should be loaded from. This is really important, especially regarding regulations of the GDPR or when developing internal systems that don't have access to the internet at all.

**Comment by rpc-scandinavia** [comment]
@chris579 you are right - especially about offline systems, MS show that you can wrap Blazor in a Electron UI in the future, and thus creating a standalone GUI application with Blazor.

The issue with the Google font, is that the icons are in the font (as I understand it). In my project I have a lot of SVG files gathered from the internet, incouding Material icons, Apple icons and Microsoft icons. All compiled as embedded resources in an assembly living on the server, where the client can request the images.

I can not use my images with MatBlazor, where icons are needed, unless I do as I have done a couple of times. I copy the "MatButton.razor" file and and put it in my project as "ElementMatButton.razor". It still inherits "BaseMatButton", the only thins I do is replacing the two "span"s with my own icon element.

I thing that MatBlazor somehow need a way to configure custom icon element in "startup.cs", and perhaps something similar in other areas, like something more easy regarding colours and typography.

Just a few thoughts.

**Comment by chris579** [comment]
> I can not use my images with MatBlazor, where icons are needed, unless I do as I have done a couple of times. I copy the "MatButton.razor" file and put it in my project as "ElementMatButton.razor". It still inherits "BaseMatButton", the only thing I do is replacing the two icon "span"s with my own icon element.
> 
> I think that MatBlazor somehow need a way to configure custom icon element in "startup.cs", and perhaps something similar in other areas, like something more easy regarding colours and typography.

I like that idea, you should create a separate issue for this.
